File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // get float input from user
    float change;
    do
    {
       change = get_float("Change owed: ");
    }
    while (change < 0.00);
    // round up to hundreds in one transaction
    int n = round(change * 100);
    
    // determining integers required
    int quarter, dime, nickel, penny, greedy;
    
    // cashier
    quarter = n / 25;
    dime = (n % 25) / 10;
    nickel = ((n % 25) % 10) / 5;
    penny = ((n % 25) % 10) % 5;
    greedy = quarter + dime + nickel + penny;
    printf("%i\n", greedy);
}

File: cash.py
from cs50 import get_float

while True:
    n = get_float("Change owed: ")
    if n >= 0:
        break

change = round(n * 100)
coin_count = 0

for i in [25, 10, 5, 1]:
    if change < i:
        continue
    else:
        coin_count += int(change / i)
        change = change % i

print(coin_count)